### ISSUE TITLE
DKG Key Refreshing via Proactive secret sharing

### ISSUE THREAD

**Issue by cygnusv** [nucypher/ferveo#3]
Related issue: https://github.com/nucypher/tdec/issues/74

**Comment by cygnusv** [comment]
See https://github.com/anoma/ferveo/issues/18

**Comment by theref** [comment]
What would trigger this? PRE nodes currently have an idea of the fleet state. Would it just be a case of monitoring that equivalent state and triggering whenever a node leaves that was part of a DKG ceremony that I participated in?

**Comment by piotr-roslaniec** [comment]
During [our code reviews](https://github.com/nucypher/ferveo/pull/14) we put forward a few ideas on how we could alter Ferveo to better fit our needs:
- Remove weighting from nodes - each node gets an equal amount of key shares
- Further, we set a limit of one share per node

With these changes in mind, it seems like the key refresh described in the original issue (https://github.com/anoma/ferveo/issues/18) could be simplified to "running the same DKG again with final key 0". 

This is not exactly the kind of solution we want ([Proactive secret sharing](https://www.wikiwand.com/en/Proactive_secret_sharing#Example)). In our approach, we would need DKG participants to construct random polynomials and use them to update their key shares. 

I suspect this work may be blocked by the [alternations we've made during our code review](https://github.com/nucypher/ferveo/pull/14/files#diff-b9683cc455c05bd6e307e7a9002573f586accc82b58602db05e96c65e150cd1dR179).
